qiskit_ibm_provider/job/ibm_circuit_job.py

```diff
@@ -681,7 +681,10 @@ def _retrieve_result(self, refresh: bool = False) -> None:
 
         if not self._result or refresh:  # type: ignore[has-type]
             try:
-                api_result = self._runtime_client.job_results(self.job_id())
+                if self._provider._runtime_client.job_type(self.job_id()) == "IQX":
+                    api_result = self._api_client.job_result(self.job_id())
+                else:
+                    api_result = self._runtime_client.job_results(self.job_id())
```

Comment:
I thought the runtime API supports getting result for the old IQX jobs?

Reply:
I don't believe so, I tested with https://runtime-us-east.quantum-computing.ibm.com/openapi#/Jobs/get_job_results_jid - the runtime api can't retrieve IQX jobs or the results 

```
404 Client Error: Not Found for url: ...
```